### PR TITLE
Add weights

### PR DESCRIPTION
**Pull request by apoorvsadana**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

-
-
-

### CODE REVIEW

**Review by EvolveArt — CHANGES_REQUESTED** [review]
left few comments

File: crates/pallets/starknet/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
...an\":{\"lhs\":{\"Deref\":{\"register\":\"FP\",\"offset\":-3}},\"rhs\":{\"Immediate\":\"0x100000000000000000000000000000000\"},\"dst\":{\"register\":\"AP\",\"offset\":0}}}]],[737,[{\"DivMod\":{\"lhs\":{\"Deref\":{\"register\":\"FP\",\"offset\":-3}},\"rhs\":{\"Immediate\":\"0x100000000000000000000000000000000\"},\"quotient\":{\"register\":\"AP\",\"offset\":3},\"remainder\":{\"register\":\"AP\",\"offset\":4}}}]],[775,[{\"TestLessThan\":{\"lhs\":{\"Deref\":{\"register\":\"AP\",\"offset\":0}},\"rhs\":{\"Immediate\":\"0x100000000000000000000000000000000\"},\"dst\":{\"register\":\"AP\",\"offset\":-1}}}]],[798,[{\"TestLessThan\":{\"lhs\":{\"Deref\":{\"register\":\"AP\",\"offset\":0}},\"rhs\":{\"Immediate\":\"0x100000000000000000000000000000000\"},\"dst\":{\"register\":\"AP\",\"offset\":-1}}}]],[818,[{\"TestLessThan\":{\"lhs\":{\"Deref\":{\"register\":\"AP\",\"offset\":0}},\"rhs\":{\"Immediate\":\"0x100000000000000000000000000000000\"},\"dst\":{\"register\":\"AP\",\"offset\":-1}}}]]],\"entry_points_by_type\":{\"EXTERNAL\":[{\"selector\":\"0x2e7e6ab3df2d8d293198339ef8cda98658e6ddc2d7ffb24116e343e26d3db8d\",\"offset\":0,\"builtins\":[\"range_check\"]}],\"L1_HANDLER\":[],\"CONSTRUCTOR\":[]}}";
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/benchmarking.rs]: ya I thought that needed `std` but it works

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -161,13 +166,22 @@ pub mod pallet {
...
         #[pallet::constant]
         type ValidateMaxNSteps: Get<u32>;
         #[pallet::constant]
         type ProtocolVersion: Get<u8>;
         #[pallet::constant]
         type ChainId: Get<Felt252Wrapper>;
+        /// Gas limit of one block.
+        #[pallet::constant]
+        type BlockGasLimit: Get<u64>;
+        /// Maps Ethereum gas to Substrate weight.
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: fixed

File: crates/pallets/starknet/src/lib.rs [hunk 3]
```diff
@@ -161,13 +166,22 @@ pub mod pallet {
...
+        type StepWeightMapping: StepWeightMapping;
+        /// Weight corresponding to a gas unit.
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: fixed

File: crates/pallets/starknet/src/lib.rs [hunk 4]
```diff
@@ -161,13 +166,22 @@ pub mod pallet {
...
+        type WeightPerStep: Get<Weight>;
+        /// Gas limit Pov size ratio.
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: fixed

File: crates/pallets/starknet/src/lib.rs [hunk 5]
```diff
@@ -715,7 +762,10 @@ pub mod pallet {
         /// # TODO
         /// * Compute weight
         #[pallet::call_index(4)]
-        #[pallet::weight({0})]
+        #[pallet::weight({
+			let without_base_extrinsic_weight = true;
+			T::StepWeightMapping::steps_to_weight(T::TransactionMaxNSteps::get(), without_base_extrinsic_weight)
+		})]
         pub fn consume_l1_message(origin: OriginFor<T>, transaction: Transaction) -> DispatchResult {
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: ya missed this one, fixed now

File: crates/pallets/starknet/src/lib.rs [hunk 6]
```diff
@@ -834,6 +884,36 @@ pub mod pallet {
     }
 }
 
+/// A mapping function that converts Starknet gas to Substrate weight
+pub trait StepWeightMapping {
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: moved it to `primitives/weights`

File: crates/pallets/starknet/src/lib.rs [hunk 7]
```diff
@@ -834,6 +884,36 @@ pub mod pallet {
     }
 }
 
+/// A mapping function that converts Starknet gas to Substrate weight
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/pallets/starknet/src/lib.rs]: fixed

**Pull request closed without merging**